src/planner/mod.rs

```diff
@@ -110,21 +122,26 @@ pub trait Planner: std::fmt::Debug + Send + Sync + dyn_clone::DynClone {
     async fn plan(&self) -> Result<Vec<StatefulAction<Box<dyn Action>>>, PlannerError>;
     /// The settings being used by the planner
     fn settings(&self) -> Result<HashMap<String, serde_json::Value>, InstallSettingsError>;
+    /// The default settings being used by the planner
+    async fn default_settings(
```

Comment:
This feels like rule breaking and I think both of us mentioned we didn't love the API...

How would you feel about making `configured_settings` something the various planners have to implement, and removing this implementation requirement? Then we can remove the generic `configured_settings` implementation and each of them can implement this:


```rust
    async fn configured_settings(
        &self,
    ) -> Result<HashMap<String, serde_json::Value>, PlannerError> {
        let default = Self::default().await?.settings()?;
        let configured = self.settings()?;

        let mut settings: HashMap<String, serde_json::Value> = HashMap::new();
        for (key, value) in configured.iter() {
            if default.get(key) != Some(value) {
                settings.insert(key.clone(), value.clone());
            }
        }

        Ok(settings)
    }
```

It seems to work for me locally.

Reply:
(Not quite sure what you mean by "rule breaking")

Yeah, that's what I had in ee7d24c968750762036a33649e40e9c4afb581b0, so I think we can just go back to it.

Reply:
Hmm, so the reason we added `default_setting(&self)` is so we could call `Self::default()` yes? Which we could not do due to object safety rules. So we're essentially skirting object safey by forcing the planner to re-implement `Self::default()` without referring to `Self`. It feels like rule breaking. :)